### ISSUE TITLE
[BUG] [Planetscale] Cannot create a JSON value from a string with CHARACTER SET 'binary'

### ISSUE THREAD

**Issue by satvikreddy** [prisma/prisma#25594]
### Bug description
**Happening in planetscale dev databases currently. Across prisma versions!!**

**Issue**
Any time i try updating a json column i get this error:
`Cannot create a JSON value from a string with CHARACTER SET 'binary'`

NOTE: this issue happens when directly using PrismaClient with planetscale.
Adding @prisma/adapter-planetscale magically resolves the issue.
I though adapter was an improved way but not the only way right? Or is it mandatory? Since the same was working for past few months without any adapter.

**Workaround**
Using planetscale prisma adapter gets it to work: https://www.npmjs.com/package/@prisma/adapter-planetscale

I can confirm that the it is not database issue as direct mysql queries work:
```
UPDATE Trip
SET customFieldValues = '{"abcde": true}'
WHERE id = 'cm2qe8r5i015tc8kjfxbzlpgk';
```

```
UPDATE Trip
SET customFieldValues = JSON_OBJECT('abcd', true)
WHERE id = 'cm2qe8r5i015tc8kjfxbzlpgk';
```

### Prisma information

```prisma
model Trip {
  id                      String     @id @default(cuid())
  createdTime             DateTime   @default(now())
  customFieldValues       Json       @default("{}")
}
```

```ts
const prisma = new PrismaClient();

await prisma.trip.update({
    where: {
      id: 'cm2qe8r5i015tc8kjfxbzlpgk',
    },
    data: {
      customFieldValues: {
        abc: true,
      },
    },
  });
```

### Environment & setup

- OS: Windows
- Database: Planetscale
- Node.js version: 20.10.0

### Prisma Version

```
5.22.0
```

**Comment by a20hek** [comment]
facing the same issue with planetscale since a few hours
have had no package updates in the last few days
Prisma version: `5.21.1`

**Comment by JayeshVP24** [comment]
Same issue, Prisma version `5.21.1`

**Comment by mtorp** [comment]
Facing the same issue. I tried rolling back to Prisma 5.18.0, but the problem persists. It also looks like it’s only affecting our test branches on PlanetScale.

**Comment by satvikreddy** [comment]
@mtorp @JayeshVP24 @a20hek 

Found this quick workaround after some trial and error. Maybe it helps you.

Following steps to add this magically gets it to work. (Don't know why though)
https://www.npmjs.com/package/@prisma/adapter-planetscale

**Comment by mtorp** [comment]
Thanks @satvikreddy!
Is the issue also only affecting your development branches?

**Comment by satvikreddy** [comment]
> Thanks @satvikreddy! Is the issue also only affecting your development branches?

Yes. Only dev branches currently.

**Comment by mtorp** [comment]
I reported the issue to PlanetScale and they acknowledge that it's a problem on their side related to an upgrade of Vitess they conducted yesterday. They are currently working on a fix.

**Comment by orware** [comment]
If you are experiencing this issue please reach out via https://planetscale.com/contact and we can help make an adjustment for your development branches in the meanwhile!

**Comment by kyujonglee** [comment]
Same issue 😥. Prisma Version: 5.21.1

Any updates?

**Comment by jferrettiboke** [comment]
Same issue here! Hope PlanetScale releases a fix quickly.

**Comment by orware** [comment]
We already have an updated build for development branches pushed out so folks should be all good again! Thank you for your patience earlier as the team worked through that on our end!

**Comment by aqrln** [comment]
Hey, sounds like this was already taken care of.

@orware thanks so much for engaging here, and please let us know if we can close this issue, or if there's anything we should also do on our side.

**Comment by orware** [comment]
@aqrln No problem! It should be safe to close on this side!

**Comment by jedi4z** [comment]
I used the planet scale connection string and it works. I experienced this error before when I used `pscale connect`

```
DATABASE_URL='mysql://<user>:<pass>@aws.connect.psdb.cloud/db?ssl={"rejectUnauthorized":true}&&sslaccept=strict'
```

**Comment by rickitan** [comment]
Weird I experienced the same as you @jedi4z 

It fails when I use `pscale connect` but works if I use the `aws` connection string.

Did you figure out a way to get if working while using `pscale connect`?

I'm not using the `@prisma/adapter-planetscale`

Pinging @orware by any chance do you know what could be happening?

**Comment by orware** [comment]
@rickitan The current behavior is likely related to using the most recent `pscale` versions, v0.250.0 and newer, where that behavior seems to have come back that is being investigated over in this issue report at the moment:
https://github.com/planetscale/cli/issues/1109#issuecomment-3230516857

With the help of another user I was able to work through creating a minimal reproduction today that I'm about to share with our team to investigate a bit further but as a quick solution you can drop down temporarily to v0.249.0 to continue using `pscale connect` until we get it resolved.

**Comment by paras-genmo** [comment]
Any updates on this? Would prefer to use more recent pscale CLI versions rather than pinning to an old version as suggested

**Comment by orware** [comment]
Hello @paras-genmo, none to report at the moment. 

We do have an internal issue report filed with our team in addition to the CLI one linked above but a resolution is still pending.

**Comment by orware** [comment]
@paras-genmo and @rickitan I have an update on the more recent occurrence for this issue as it should hopefully be addressed now with v0.267.0 or newer of the `pscale` CLI:
https://github.com/planetscale/cli/issues/1109#issuecomment-3633651240
